pandas/tests/groupby/aggregate/test_other.py

```diff
@@ -512,3 +512,20 @@ def test_agg_category_nansum(observed):
     if observed:
         expected = expected[expected != 0]
     tm.assert_series_equal(result, expected)
+
+
+def test_agg_list_like_func():
+    # GH 18473
+    df = pd.DataFrame({'A': [str(x) for x in range(10)],
+                       'B': [str(x) for x in range(10)]})
+    result = df.groupby('A', as_index=False, sort=False)\
+               .aggregate({'B': lambda x: list(x)})
+    expected = df.copy()
+    expected['B'] = df['B'].apply(lambda x: list(x))
```

Comment:
Would be preferable to just manually construct the expected output here for both readability and risk mitigation. A smaller example (say `range(3)`) should suffice if it makes things easier